app/controllers/api/subcollections/tags.rb

```diff
@@ -84,11 +93,8 @@ def parse_tag(data)
       end
 
       def parse_tag_from_href(data)
-        href = data["href"]
-        tag  = if href && href.match(%r{^.*/tags/#{CID_OR_ID_MATCHER}$})
-                 klass = collection_class(:tags)
-                 klass.find(from_cid(href.split('/').last))
-               end
+        href = parse_href(data['href'])
```

Comment:
this code should only be defining tag if the href is indeed for a tag, so we can't just simply use parse_href. the code can be generalized if this is needed somewhere where where we send in the collection of interest, (i.e. :tags) then that drives the generalization of the old lines 88-91.  Not sure where else we do this though.